### PR TITLE
Release 3.1.0

### PR DESCRIPTION
**Pull request by smk4664**
### Added

- [#98](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/98) - Python 3.12 support

### Changed

- [#84](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/84) - Replaced pydocstyle with ruff.

### Fixed

- [#92](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/92) - Changed Site to Location per model updates in Nautobot 2.0.

### Housekeeping

- [#8](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/8) - Re-baked from the latest template.
- [#96](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/96) - Rebaked from the cookie `nautobot-app-v2.3.0`.
- [#98](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/98) - Rebaked from the cookie `nautobot-app-v2.3.2`.

### CODE REVIEW

File: docs/admin/release_notes/version_3.1.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# v3.1 Release Notes
+
+This document describes all new features and changes in the release `3.1`. The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [v3.1.0 (2024-08-27)](https://github.com/nautobot/nautobot-app-capacity-metrics/releases/tag/v3.1.0)
+
+### Added
+
+- [#98](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/98) - Python 3.12 support
```
Comment (gsnider2195) [inline docs/admin/release_notes/version_3.1.md:9]:
```suggestion
- [#98](https://github.com/nautobot/nautobot-app-capacity-metrics/issues/98) - Added support for Python 3.12.
```

File: docs/admin/compatibility_matrix.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Compatibility Matrix
+
+| Metrics & Monitoring Extension App Version | Nautobot First Support Version | Nautobot Last Support Version |
+| ------------- | -------------------- | -------------- |
+| 1.0.X         | 1.0.0                | 1.99.99        |
+| 1.1.X         | 1.0.0                | 1.99.99        |
+| 2.0.X         | 1.2.0                | 1.99.99        |
+| 2.1.X         | 1.6.0                | 1.99.99        |
+| 3.0.X         | 2.0.0                | 2.99.99        |
```
Comment (gsnider2195) [inline docs/admin/compatibility_matrix.md:9]:
```suggestion
| 3.0.X         | 2.0.0                | 2.99.99        |
| 3.1.X         | 2.0.0                | 2.99.99        |
```

File: .bandit.yml [hunk 3]
```diff
@@ -3,3 +3,4 @@ skips: []
 # No need to check for security issues in the test scripts!
```
Comment (gsnider2195) [inline .bandit.yml:3]: This file and `.flake8` should have been deleted correct?

Comment (smk4664) [reply .bandit.yml:3]: That wasn't done as part of the drift manager. It was the first one that I touched the first round/

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,122 +1,152 @@
...
       INVOKE_NAUTOBOT_CAPACITY_METRICS_LOCAL: "True"
     steps:
       - name: "Check out repository code"
-        uses: "actions/checkout@v2"
+        uses: "actions/checkout@v4"
       - name: "Setup environment"
-        uses: "networktocode/gh-action-setup-poetry-environment@v5"
+        uses: "networktocode/gh-action-setup-poetry-environment@v6"
       - name: "Linting: bandit"
         run: "poetry run invoke bandit"
```
Comment (gsnider2195) [inline .github/workflows/ci.yml:40]: Looks like we need to remove this

**Pull request merged**